src/libraries/System.Text.Json/tests/System.Text.Json.SourceGeneration.Tests/JsonSourceGeneratorTests.cs

```diff
@@ -466,5 +466,19 @@ public class ClassWithEnumAndNullable
             public DayOfWeek Day { get; set; }
             public DayOfWeek? NullableDay { get; set; }
         }
+
```

Comment:
Do we have a test for `JsonSerializerContext.GetTypeInfo(type)` that ensures all converters are not rooted? Or is this self-evident by other context tests?

Reply:
I've augmented the tests to simulate this via reflection, since the test projects are currently not trimmed. I'll chat with @joperezr offline about a test plan where we source-gen and run the linker.